controllers/front/ListComments.php

```diff
@@ -64,17 +64,24 @@ public function display()
             $productComment['content'] = htmlentities($productComment['content']);
             $productComment['date_add'] = $dateFormatter->format($dateAdd);
 
-            if ($isLastNameAnonymous && isset($productComment['lastname'])) {
-                $productComment['lastname'] = substr($productComment['lastname'], 0, 1) . '.';
-            }
+            // The customer has firstname and lastname, for guest we only have customer_name field
+            $productComment['customer_name'] = !empty($productComment['customer_name'])
+                ? $productComment['customer_name']
+                : $productComment['firstname'] . ' ' . $productComment['lastname'];
 
-            // if registered customer : return customer first and last name instead of using customer_name
-            if (!empty($productComment['lastname'])) {
-                $productComment['customer_name'] = htmlentities($productComment['firstname'] . ' ' . $productComment['lastname']);
-            } else {
-                $productComment['customer_name'] = htmlentities($productComment['customer_name']);
+            if ($isLastNameAnonymous) {
+                $parts = explode(' ', $productComment['customer_name']);
+                $guestFirstname = $parts[0];
+                $productComment['firstname'] = $guestFirstname;
+                $productComment['lastname'] = count($parts) > 1 ? array_pop($parts) : '';
```

Comment:
I think `$productComment['firstname'] = $parts[0];` will be better.
Because this variable is used only one time and its name looks confused.